collector-http/src/test/java/zipkin2/collector/otel/http/ITLogEventEmission.java

```diff
@@ -0,0 +1,421 @@
+      assertThat(spans.get(1).tags()).containsKey("otel.scope.version");
+      assertThat(spans.get(1).tags()).containsEntry("telemetry.distro.name", "elastic");
+      assertThat(spans.get(1).tags()).containsKey("telemetry.distro.version");
+      assertThat(spans.get(1).tags()).containsEntry("telemetry.sdk.language", "nodejs");
+      assertThat(spans.get(1).tags()).containsEntry("telemetry.sdk.name", "opentelemetry");
+      assertThat(spans.get(1).tags()).containsKey("telemetry.sdk.version");
+      assertThat(spans.get(2).id()).isEqualTo(spanId);
+      assertThat(spans.get(2).traceId()).isEqualTo(traceId);
+      assertThat(spans.get(2).annotations()).isEmpty();
+      assertThat(spans.get(2).kind()).isEqualTo(Span.Kind.CLIENT);
+      assertThat(spans.get(2).name()).isEqualTo("chat gpt-4o-mini");
+      assertThat(spans.get(2).duration()).isGreaterThan(0);
+      assertThat(spans.get(2).localEndpoint()).isNotNull();
+      assertThat(spans.get(2).localEndpoint().serviceName()).isEqualTo("opentelemetry-nodejs-openai");
+      assertThat(spans.get(2).tags()).hasSize(17);
+      assertThat(spans.get(2).tags()).containsEntry("gen_ai.operation.name", "chat");
+      assertThat(spans.get(2).tags()).containsEntry("gen_ai.request.model", "gpt-4o-mini");
+      assertThat(spans.get(2).tags()).containsEntry("gen_ai.response.finish_reasons", "stop");
+      assertThat(spans.get(2).tags()).containsEntry("gen_ai.response.id", "chatcmpl-1234");
+      assertThat(spans.get(2).tags()).containsEntry("gen_ai.response.model", "gpt-4o-mini");
+      assertThat(spans.get(2).tags()).containsEntry("gen_ai.system", "openai");
+      assertThat(spans.get(2).tags()).containsEntry("gen_ai.usage.input_tokens", "5");
+      assertThat(spans.get(2).tags()).containsEntry("gen_ai.usage.output_tokens", "7");
+      assertThat(spans.get(2).tags()).containsEntry("otel.scope.name", "@elastic/opentelemetry-instrumentation-openai");
+      assertThat(spans.get(2).tags()).containsKey("otel.scope.version");
+      assertThat(spans.get(2).tags()).containsEntry("server.address", "host.testcontainers.internal");
+      assertThat(spans.get(2).tags()).containsEntry("server.port", String.valueOf(openAiPort));
+      assertThat(spans.get(2).tags()).containsEntry("telemetry.distro.name", "elastic");
+      assertThat(spans.get(2).tags()).containsKey("telemetry.distro.version");
+      assertThat(spans.get(2).tags()).containsEntry("telemetry.sdk.language", "nodejs");
+      assertThat(spans.get(2).tags()).containsEntry("telemetry.sdk.name", "opentelemetry");
+      assertThat(spans.get(2).tags()).containsKey("telemetry.sdk.version");
+    }
+  }
+
+  static class ChatCompletionService {
+    @Post("/v1/chat/completions")
+    @Consumes("application/json")
+    @Produces("application/json")
+    public Map<String, Object> handleChatCompletion(String requestBody) {
+      Map<String, Object> response = new HashMap<>();
```

Comment:
Not sure it works, but prefer `Map.of` stuff if it is usable. that or just a multi-line json literal. We can use a later JDK in tests than we do for the compiled binary (floor JRE version). This can be done later.